### PR TITLE
[Docs] Refine Quick Start

### PR DESCRIPTION
**Pull request by fanqiNO1**

### CODE REVIEW

File: docs/zh_cn/get_started/quickstart.md [hunk 1]
```diff
@@ -1,3 +1,263 @@
...
+
+在合并权重后，为了更好地体会到模型的能力，XTuner 也集成了与合并后的模型对话的功能。通过如下命令，便可以启动一个与模型对话的简易 Demo。
+
+```bash
+xtuner chat work_dirs/internlm2_7b_qlora_colorist_e5_copy/merged\
+                --prompt-template internlm2_chat\
+                --system-template colorist
+```
+
+其中 `work_dirs/internlm2_7b_qlora_colorist_e5_copy/merged` 是合并后的权重路径，`--prompt-template internlm2_chat` 指定了对话模板为 InternLM2-Chat，`--system-template colorist` 则是指定了与模型对话时的 System Prompt 为colorist 数据集所要求的模板。
```
Comment (Gennadiyev) [inline docs/zh_cn/get_started/quickstart.md]: `为colorist 数据集所要求的模板。` -> `为 Colorist 数据集所要求的模板。`

File: docs/zh_cn/get_started/quickstart.md [hunk 2]
```diff
@@ -1,3 +1,263 @@
...
+
+由于相同的问题，因此我们从 ModelScope 下载所需要的微调数据集。
+
+```bash
+git clone https://www.modelscope.cn/datasets/fanqiNO1/colors.git
+```
+
+## 准备配置文件
+
+XTuner 提供了多个开箱即用的配置文件，可以通过 xtuner list-cfg 查看。我们执行如下指令，以复制一个配置文件到当前目录。
```
Comment (Gennadiyev) [inline docs/zh_cn/get_started/quickstart.md]: xtuner list-cfg 周围应有 <code>`</code> 标识。

File: docs/zh_cn/get_started/quickstart.md [hunk 3]
```diff
@@ -1,3 +1,263 @@
...
+<s><|im_start|>system
+You are a professional color designer. Please provide the corresponding colors based on the description of Human.
+<|im_end|>
+<|im_start|>user
+Please give me a clear blue like the sky.<|im_end|>
+<|im_start|>assistant
+The color you're describing is a clear, bright blue that resembles the color of the sky on a sunny day. This color is often referred to as "sky blue" or "cerulean blue." Here are some specific color codes and names that
+```
+
+而在训练完全完成后，可以观察到模型的输出是这样的：
```
Comment (Gennadiyev) [inline docs/zh_cn/get_started/quickstart.md]: 建议 `是这样的` -> `如下所示`，出于文档文本严谨性考量。上、下同。

**Review by Gennadiyev — COMMENTED** [review]
I appreciate the efforts made in drafting this documentation & would like to suggest several minor format-related changes to the documentation (see details).

I suggest inserting external website URLs to this quickstart to ensure a more informative presentation in favor of the users. ModelScope and HuggingFace should be have hyperlinks to the corresponding model checkpoints to provide a seamless experience to the end developers.

Looking forward to seeing these adjustments incorporated, and I'm here to assist further if needed!

File: docs/en/get_started/quickstart.md [hunk 4]
```diff
@@ -1,3 +1,275 @@
 # Quickstart
 
-In this section, we will show you how to use XTuner to tune a model.
+In this section, we will show you how to use XTuner to fine-tune a model to help you get started quickly.
+
+After installing XTuner successfully, we can start fine-tuning the model. In this section, we will demonstrate how to use XTuner to apply the QLoRA algorithm to fine-tune InternLM2-Chat-7B on the Colorist dataset.
+
+The Colorist dataset is a dataset that provides color choices and suggestions based on color descriptions. A model fine-tuned on this dataset can be used to give a hexadecimal color code based on the user's description of the color. For example, when the user enters "a calming but fairly bright light sky blue, between sky blue and baby blue, with a hint of fluorescence due to its brightness", the model will output ![#66ccff](https://img.shields.io/badge/%2366ccff-66CCFF), which matches the user's description. There are a few sample data from this dataset:
```
Comment (LZHgrla) [inline docs/en/get_started/quickstart.md]: Please add dataset URL here.

File: docs/en/get_started/quickstart.md [hunk 5]
```diff
@@ -1,3 +1,275 @@
...
+| Bright red: This is a very vibrant, saturated and vivid shade of red, resembling the color of ripe apples or fresh blood. It is as red as you can get on a standard RGB color palette, with no elements of either blue or green. | 鲜红色： 这是一种非常鲜艳、饱和、生动的红色，类似成熟苹果或新鲜血液的颜色。它是标准 RGB 调色板上的红色，不含任何蓝色或绿色元素。 | #ee0000: ![#ee0000](https://img.shields.io/badge/%23ee0000-EE0000) |
+| Bright Turquoise: This color mixes the freshness of bright green with the tranquility of light blue, leading to a vibrant shade of turquoise. It is reminiscent of tropical waters.                                              | 明亮的绿松石色：这种颜色融合了鲜绿色的清新和淡蓝色的宁静，呈现出一种充满活力的绿松石色调。它让人联想到热带水域。                 | #00ffcc: ![#00ffcc](https://img.shields.io/badge/%2300ffcc-00FFCC) |
+
+## Prepare the model weights
+
+Before fine-tuning the model, we first need to prepare the weights of the model. Since XTuner pulls model weights from HuggingFace by default, which may lead to an unstable download process, slow download speed and other problems, we choose to download the weights of InternLM2-Chat-7B from ModelScope.
```
Comment (LZHgrla) [inline docs/en/get_started/quickstart.md]: Please give two download examples. The first is from HuggingFace, and the second is from ModelScope.
Specify that only users experiencing network issues should consider using ModelScope.

File: docs/en/get_started/quickstart.md [hunk 6]
```diff
@@ -1,3 +1,275 @@
...
+- The HuggingFace link is located at: https://huggingface.co/internlm/internlm2-chat-7b
+- The ModelScope link is located at: https://modelscope.cn/models/Shanghai_AI_Laboratory/internlm2-chat-7b/summary
+
+## Prepare the fine-tuning dataset
+
+Due to the same reason, we choose to download the dataset from ModelScope.
+
+```bash
+git clone https://www.modelscope.cn/datasets/fanqiNO1/colors.git
+```
```
Comment (LZHgrla) [inline docs/en/get_started/quickstart.md:63]: Please give two examples, one is from HuggingFace and another one is from ModelScope.

File: docs/en/get_started/quickstart.md [hunk 7]
```diff
@@ -1,3 +1,275 @@
...
+    dataset=dict(type=load_dataset, path='json', data_files=dict(train=data_path)),
+    tokenizer=tokenizer,
+    max_length=max_length,
+    dataset_map_fn=colors_map_fn,
+    template_map_fn=dict(
+        type=template_map_fn_factory, template=prompt_template),
+    remove_unused_columns=True,
+    shuffle_before_pack=True,
+    pack_to_max_length=pack_to_max_length)
+```
```
Comment (LZHgrla) [inline docs/en/get_started/quickstart.md]: Please use `diff` markdown block (like [here](https://github.com/InternLM/xtuner/tree/main/examples/demo_data/multi_turn_2#config)), instead of `python + comment`

File: docs/en/get_started/quickstart.md [hunk 8]
```diff
@@ -1,3 +1,275 @@
...
+Once having done the above steps, we can start fine-tuning using the following command.
+
+```bash
+# Single GPU
+xtuner train internlm2_7b_qlora_colorist_e5_copy.py
+# Multiple GPUs
+NPROC_PER_NODE=${GPU_NUM} xtuner train internlm2_7b_qlora_colorist_e5_copy.py
+# Slurm
+srun ${SRUN_ARGS} xtuner train internlm2_7b_qlora_colorist_e5_copy.py --launcher slurm
+```
```
Comment (LZHgrla) [inline docs/en/get_started/quickstart.md]:
```suggestion
```bash
# Single GPU
xtuner train ./internlm2_7b_qlora_colorist_e5_copy.py
# Multiple GPUs
NPROC_PER_NODE=${GPU_NUM} xtuner train ./internlm2_7b_qlora_colorist_e5_copy.py
# Slurm
srun ${SRUN_ARGS} xtuner train ./internlm2_7b_qlora_colorist_e5_copy.py --launcher slurm
```
```

File: docs/en/get_started/quickstart.md [hunk 9]
```diff
@@ -1,3 +1,275 @@
...
+The color of the model output is shown below:
+
+- 天空一样清澈透明的蓝色：![天空一样清澈透明的蓝色](https://img.shields.io/badge/天空一样清澈透明的蓝色-1099EE)
+- A clear blue like the sky: ![A clear blue like the sky](https://img.shields.io/badge/A_clear_blue_like_the_sky-0066DD)
+
+It is clear that the output of the model after training has been fully aligned with the content of the dataset.
+
+# Model Convert + LoRA Merge
+
+After training, we will get several pth files that do **not** contain all the parameters of the model, but store the parameters updated by the training process of the QLoRA algorithm. Therefore, we need to convert these pth files to hf format and merge them into the original weights.
```
Comment (LZHgrla) [inline docs/en/get_started/quickstart.md]:
```suggestion
After training, we will get several `.pth` files that do **NOT** contain all the parameters of the model, but store the parameters updated by the training process of the QLoRA algorithm. Therefore, we need to convert these `.pth` files to HuggingFace format and merge them into the original LLM weights.
```

File: docs/en/get_started/quickstart.md [hunk 10]
```diff
@@ -1,3 +1,275 @@
...
+
+### Model Convert
+
+XTuner has already integrated the function of converting the model to hf format. We can use the following command to convert the model.
```
Comment (LZHgrla) [inline docs/en/get_started/quickstart.md]:
```suggestion
XTuner has already integrated the tool of converting the model to HuggingFace format. We can use the following command to convert the model.
```

File: docs/en/get_started/quickstart.md [hunk 11]
```diff
@@ -1,3 +1,275 @@
...
+
+```bash
+# Create the directory to store parameters in hf format
+mkdir work_dirs/internlm2_7b_qlora_colorist_e5_copy/iter_720_hf
+
+# Convert the model to hf format
+xtuner convert pth_to_hf internlm2_7b_qlora_colorist_e5_copy.py\
+                            work_dirs/internlm2_7b_qlora_colorist_e5_copy/iter_720.pth\
+                            work_dirs/internlm2_7b_qlora_colorist_e5_copy/iter_720_hf
```
Comment (LZHgrla) [inline docs/en/get_started/quickstart.md]:
```suggestion
xtuner convert pth_to_hf internlm2_7b_qlora_colorist_e5_copy.py \
                            work_dirs/internlm2_7b_qlora_colorist_e5_copy/iter_720.pth \
                            work_dirs/internlm2_7b_qlora_colorist_e5_copy/iter_720_hf
```

File: docs/en/get_started/quickstart.md [hunk 12]
```diff
@@ -1,3 +1,275 @@
...
+```
+
+This command will convert `work_dirs/internlm2_7b_qlora_colorist_e5_copy/iter_720.pth` to hf format based on the contents of the config `internlm2_7b_qlora_colorist_e5_copy/iter_720.pth` and will save it in `work_dirs/internlm2_7b_qlora_colorist_e5_copy/iter_720_hf`.
```
Comment (LZHgrla) [inline docs/en/get_started/quickstart.md]:
```suggestion
This command will convert `work_dirs/internlm2_7b_qlora_colorist_e5_copy/iter_720.pth` to hf format based on the contents of the config `internlm2_7b_qlora_colorist_e5_copy.py` and will save it in `work_dirs/internlm2_7b_qlora_colorist_e5_copy/iter_720_hf`.
```

File: docs/en/get_started/quickstart.md [hunk 13]
```diff
@@ -1,3 +1,275 @@
...
+
+### LoRA Merge
+
+XTuner has also integrated the function of merging weights, we just need to execute the following command:
```
Comment (LZHgrla) [inline docs/en/get_started/quickstart.md]:
```suggestion
XTuner has also integrated the tool of merging LoRA weights, we just need to execute the following command:
```

File: docs/en/get_started/quickstart.md [hunk 14]
```diff
@@ -1,3 +1,275 @@
...
+
+```bash
+# Create the directory to store the merged weights
+mkdir work_dirs/internlm2_7b_qlora_colorist_e5_copy/merged
+
+# Merge the weights
+xtuner convert merge Shanghai_AI_Laboratory/internlm2-chat-7b\
+                        work_dirs/internlm2_7b_qlora_colorist_e5_copy/iter_720_hf\
+                        work_dirs/internlm2_7b_qlora_colorist_e5_copy/merged\
+                        --max-shard-size 2GB
```
Comment (LZHgrla) [inline docs/en/get_started/quickstart.md]:
```suggestion
xtuner convert merge Shanghai_AI_Laboratory/internlm2-chat-7b \
                        work_dirs/internlm2_7b_qlora_colorist_e5_copy/iter_720_hf \
                        work_dirs/internlm2_7b_qlora_colorist_e5_copy/merged \
                        --max-shard-size 2GB
```

File: docs/en/get_started/quickstart.md [hunk 15]
```diff
@@ -1,3 +1,275 @@
...
+```
+
+Similar to the command above, this command will read the original parameter path `Shanghai_AI_Laboratory/internlm2-chat-7b` and the path of parameter which has been converted to hf format `work_dirs/internlm2_7b_qlora_colorist_e5_copy/iter_720_hf` and merge the two parts of the parameters and save them in `work_dirs/internlm2_7b_qlora_colorist_e5_copy/merged`, where the maximum file size for each parameter slice is 2GB.
+
+## Chat with the model
+
+To better appreciate the model's capabilities after merging the weights, we can chat with the model. XTuner also integrates the function of chatting with the merged model. We can start a simple demo to chat with the model with the following command:
```
Comment (LZHgrla) [inline docs/en/get_started/quickstart.md]:
```suggestion
To better appreciate the model's capabilities after merging the weights, we can chat with the model. XTuner also integrates the tool of chatting with models. We can start a simple demo to chat with the model with the following command:
```

File: docs/en/get_started/quickstart.md [hunk 16]
```diff
@@ -1,3 +1,275 @@
...
+
+```bash
+xtuner chat work_dirs/internlm2_7b_qlora_colorist_e5_copy/merged\
+                --prompt-template internlm2_chat\
+                --system-template colorist
```
Comment (LZHgrla) [inline docs/en/get_started/quickstart.md]:
```suggestion
xtuner chat work_dirs/internlm2_7b_qlora_colorist_e5_copy/merged \
                --prompt-template internlm2_chat \
                --system-template colorist
```

File: docs/en/get_started/quickstart.md [hunk 17]
```diff
@@ -1,3 +1,275 @@
...
+```
+
```
Comment (LZHgrla) [inline docs/en/get_started/quickstart.md:295]: Add an example of chatting with a LLM and LoRA adapter here will be better!

File: docs/zh_cn/get_started/quickstart.md [hunk 18]
```diff
@@ -1,3 +1,275 @@
...
+```
+
+因此在这一步中，修改了 `pretrained_model_name_or_path`、`data_path`、`prompt_template` 以及 `train_dataset` 中的 `dataset` 字段。
+
+## 启动微调
+
+在完成上述操作后，便可以使用下面的指令启动微调任务了。
+
+```bash
+# 单张 GPU
```
Comment (LZHgrla) [inline docs/zh_cn/get_started/quickstart.md]:
```suggestion
# 单机单卡
```

File: docs/zh_cn/get_started/quickstart.md
Comment (LZHgrla) [inline docs/zh_cn/get_started/quickstart.md:1]: 请参考 en 版本 comments 进行修改。

**Review by LZHgrla — COMMENTED** [review]
Thanks for your contribution.

I left some comments, please assess if they are appropriate, and feel free to contact me if you need any help!

**Comment by fanqiNO1** [comment]
Thanks, I will fix it ASAP.

**Comment by fanqiNO1** [comment]
I have fixed it, but it seems that the failure of lint is not my fault. :joy:
